### PR TITLE
Issue #15218: formatted input files for section 4.1.3 empty blocks may be concise to follow google style guide rules

### PR DESCRIPTION
**Pull request by Zopsss**
#15218 

https://checkstyle.org/google_style.html#a4.1.3

>[EmptyBlock](https://checkstyle.org/checks/blocks/emptyblock.html#EmptyBlock)
[EmptyCatchBlock](https://checkstyle.org/checks/blocks/emptycatchblock.html#EmptyCatchBlock)

### CODE REVIEW

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java [hunk 1]
```diff
@@ -5,99 +5,92 @@
 
 import java.io.IOException;
 
-public class InputEmptyBlocksAndCatchBlocksNoViolations
-{
```
Comment (rnveach) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:9]: I assume the formatter is changing this. Is it possible to force the `{` to stay on this line like if we added a comment on the end of the previous line or made the class name excessively long?

I don't want any changes from this one. It is more a question right now.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java [hunk 2]
```diff
@@ -5,99 +5,92 @@
...
-        {
-            return;
-        }
+  public void testTryCatch() {
+    try {
+      int y = 0;
+      int u = 8;
+      int e = u - y;
+      return;
+    } catch (Exception e) {
```
Comment (rnveach) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: Why was this changed and does our google config agree? (Line separation removed)
Same with finally.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java [hunk 3]
```diff
@@ -5,99 +5,92 @@
...
-        int u=8;
-        try {
-            int e=u-y;
-        }
-        catch (IllegalArgumentException e) {
-            System.identityHashCode(e);
-            return;
-        }
+  public void testTryCatch4() {
+    int y = 0;
```
Comment (rnveach) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: Why was this changed and does our google config agree? (spaces added)

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java [hunk 4]
```diff
@@ -5,99 +5,92 @@
...
-            }
-        }
+  }
+
+  public void setFormats() {
+    try {
+      int k = 4;
+    } catch (Exception e) {
+      Object k = null;
+      if (k != null) k = "ss";
```
Comment (rnveach) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: Why was this changed and does our google config agree? (new line removed)

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 1 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:9]: >Is it possible to force the `{` to stay on this line like if we added a comment on the end of the previous line

Yes, in this case the `{` stays on the new line even after formatting though the `{` indentation increases by 1:

```java
public class InputEmptyBlocksAndCatchBlocksNoViolations // testing
 {
```

> or made the class name excessively long?

In this case, the formatter adds a line break after `public` and keeps the `{` on the same line:

```java
public
class InputEmptyBlocksAndCatchBlocksNoViolationssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss {
```

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: formatter changed it automatically, in google style guide we follow 2 spaces indentation so formatter fixed it and yes our google config agrees with this. I tried checking this with `verifyWithWholeConfig` method and it wasn't giving error for the indentation.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: formatter changed it and I tried checking the code against `verifyWithWholeConfig`, it was giving `61:7: 'if' construct must use '{}'s.` error for line 61

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: this could a bug in google formatter maybe? Cuz it is mentioned in style guide that we must use `{}`.

documentation link: https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.1.1-braces-always-used

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 1 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:9]: Formatter is not configurable. 
in this section Inputs we do not care about `{` location so we can follow formatter to be "google style friendly by all means"

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: @rnveach, if you view PR in "hide whitespaces", diff is more reasonable and at this line it just put `{` on same line as method signature.

**Review by romani — APPROVED** [review]
ok to merge

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: we are not applying Whole google yet, we just do auto format and keep chapter related Checks only.

applying whole config will be separate iteration of updates.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: None of the replies is based on what I was commenting about.

Original code was: `int y=0;` (no spaces)
New code is: `int y = 0;` (with spaces)

My comments are not about if we are following google config, my questions are we checking the changes requested by Google's formatter is the same thing our google config also complains about. My approval to use the formatter was that we would review what it is doing. It does not matter if it changes things unrelated to the test. If we don't review what it is doing, then we don't gain any information on if we can actually rely on the tool. This is why I was saying this should be its own project.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: Attention to `Braces are used`. It never says braces are optional. This looks like a bug in the formatter unless some other specific rule in the style guide says.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: See other comment.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: verifyWithWholeConfig is not yet used. It will be separate updates. 
For now we just formatting and follow formatting if that is not focus of chapter.

Example when formatter change what was intended to be correct https://github.com/checkstyle/checkstyle/pull/15301#discussion_r1682926160 and such formatting is not ok, because in that Input it matters.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: I guess there isnt other rule which enforces to not use braces

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: fact that we do not use whole config yet:
https://github.com/checkstyle/checkstyle/blob/9ee77481f67b40edd41e84e2dccd5ce5ae7cd897/src/it/java/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/EmptyBlocksMayBeConciseTest.java#L39-L42

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: > My approval to use the formatter was that we would review what it is doing.

lets merge it.
For now we see some not ideal formatting, but this is not our style, so it is fine. I have not see absolute detects of formatter. But will more clear as we start enforcing Whole config, this is a time when all problems popup and we have to deal with them.

> If we don't review what it is doing,

we do review, and we are ok only if it is unrelated to validation.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: @Zopsss , please recheck this diff line, whole exception block is gone, not expected.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: in diff without spaces:
image

looks good.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: https://google.github.io/styleguide/javaguide.html#s4.1-braces
 
I pretty sure they use this line

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: but finishing discussion on spacing should not block this PR.

**Review by romani — APPROVED** [review]
I reviewed it one more time.
.
I am ok to merge this

**Review by rdiachenko — APPROVED** [review]
lgtm

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: Nothing has changed my comments at https://github.com/checkstyle/checkstyle/pull/15289#discussion_r1683017125

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: It seems the only way to get things attention is to block PRs. It has been mentioned in discord something was an issue days ago. I have not seen the issue created for it. I question if we are just doing things blindly and not doing any real analysis. Quality should matter more than just getting the word done.

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: > @Zopsss , please open issue formatter to get 100% clarification.

I did created test cases for the highlighted part in one of my old PR. Here's that PR: https://github.com/checkstyle/checkstyle/pull/14912

We're following the rule properly and testing every cases of it

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: > It seems the only way to get things attention is to block PRs. It has been mentioned in discord something was an issue days ago. I have not seen the issue created for it. I question if we are just doing things blindly and not doing any real analysis. Quality should matter more than just getting the word done.

Opened a new issue for `{` placement as you requested: https://github.com/checkstyle/checkstyle/issues/15324 @rnveach

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: > my questions are we checking the changes requested by Google's formatter is the same thing our google config also complains about.

As of now, we're checking the input files against only some specific modules we're not checking the input against whole config. So it is hard to find if the code formatted by the formatter is always correct. If I find some bug I report it in discord or in my PR ( I will also now start opening issues for it ). But in this approach we might miss other bugs of the formatter.

When we will migrate to whole config, we will be able to find bugs more easily and I will also manually inspect the code against the style guide to ensure the rules are being followed properly. 

Yes, I agree that this method is full of flaws, we might miss bugs and all but the formatting of input files is almost complete. So for now, I think this is the best we can do.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: @rnveach, this is known issue in formatter - https://github.com/google/google-java-format/issues/51

4.1.3 section should not care about this and we are ok to reuse auto format there.

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: @Zopsss  please add to checkstyle/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java `if (k != null) k = "ss";` , in separate PR,  to triple confirm that violation still there.

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: done https://github.com/checkstyle/checkstyle/pull/15333/

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: > Why was this changed and does our google config agree? (Line separation removed) Same with finally.

yes our google config agrees with this change. I checked it with verifyWithWholeConfig method locally, and it is not giving error for having `catch (Exception e) {` on the same line as `}`, same with the finally.

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: > @Zopsss , please recheck this diff line, whole exception block is gone, not expected.

the exception block is still there:

https://github.com/checkstyle/checkstyle/blob/6f8592f629c8a398b11eeb69d39f5f619904c8b5/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java#L23-L26

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: Confirmed, there is catch.
Strange, my bad.

I am fully satisfied on this review point

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: Merged

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: > does our google config agree? (spaces added)

yes it does, it also gives violations if `=` does not have spaces around it:

```java
/** Some javadoc. */
public class Test {
  public void testing() {
    int x=10;
    int y = 10;
  }
}
```

```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\OpeningCurlyBracesPlacement.java
Starting audit...
[WARN] Test.java:4:10: WhitespaceAround: '=' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] Test.java:4:10: WhitespaceAround: '=' is not preceded with whitespace. [WhitespaceAround]
Audit done.
```

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: confirming that our google config agrees with the formatted code:

```java
/** Some javadoc. */
public class Test {
  /** Some javadoc. */
  public void testTryCatch3() {
    try {
      int y = 0;
      int u = 8;
      int e = u - y;
    } catch (IllegalArgumentException e) {
      System.identityHashCode(e); // some comment
      return;
    } catch (IllegalStateException ex) {
      System.identityHashCode(ex);
      return;
    }
  }
}
```

```
java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\Test.java
Starting audit...
Audit done.
```

Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: > confirming that our google config agrees with the formatted code

This just shows that our google config does not put any violations on the "formatted" code. Confirmation requires that our google config would violate the original code and require us to change it to this new version.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 3 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:46]: I am good here. Old code is violated, and requires "formatted" code.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: This still needs a response.

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: @rnveach , 
I am not sure what is left unanswered.

We found issue https://github.com/google/google-java-format/issues/51 it is known defect 

And we checkstyle violate this code, it is good - confirmed in that issue.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]:
```java
/** Some javadoc. */
public class Test {
  /** Some javadoc. */
  public void testTryCatch3() {
    // formatted code
    try {
      int y = 0;
      int u = 8;
      int e = u - y;
    } catch (IllegalArgumentException e) {
      System.identityHashCode(e); // some comment
      return;
    } catch (IllegalStateException ex) {
      System.identityHashCode(ex);
      return;
    }

    // original code
    try {
        int y=0;
        int u=8;
        int e=u-y;
        return;
    }
    catch (Exception e) {
        System.identityHashCode(e);
        return;
    }
    finally
    {
        return;
    }
  }
}
```

```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\Test.java                       
Starting audit...
[WARN] \Test.java:18:9: 'try' child has incorrect indentation level 8, expected level should be 6. [Indentation]
[WARN] \Test.java:18:14: WhitespaceAround: '=' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] \Test.java:18:14: WhitespaceAround: '=' is not preceded with whitespace. [WhitespaceAround]
[WARN] \Test.java:19:9: 'try' child has incorrect indentation level 8, expected level should be 6. [Indentation]
[WARN] \Test.java:19:14: WhitespaceAround: '=' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] \Test.java:19:14: WhitespaceAround: '=' is not preceded with whitespace. [WhitespaceAround]
[WARN] \Test.java:20:9: 'try' child has incorrect indentation level 8, expected level should be 6. [Indentation]
[WARN] \Test.java:20:14: WhitespaceAround: '=' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] \Test.java:20:14: WhitespaceAround: '=' is not preceded with whitespace. [WhitespaceAround]
[WARN] \Test.java:20:16: WhitespaceAround: '-' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] \Test.java:20:16: WhitespaceAround: '-' is not preceded with whitespace. [WhitespaceAround]
[WARN] \Test.java:21:9: 'try' child has incorrect indentation level 8, expected level should be 6. [Indentation]
[WARN] \Test.java:22:5: '}' at column 5 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurlySame]
[WARN] \Test.java:24:9: 'catch' child has incorrect indentation level 8, expected level should be 6. [Indentation]
[WARN] \Test.java:25:9: 'catch' child has incorrect indentation level 8, expected level should be 6. [Indentation]
[WARN] \Test.java:26:5: '}' at column 5 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurlySame]
[WARN] \Test.java:28:5: '{' at column 5 should be on the previous line. [LeftCurly]
[WARN] \Test.java:28:5: 'finally lcurly' has incorrect indentation level 4, expected level should be 6. [Indentation]
[WARN] \Test.java:30:5: 'finally rcurly' has incorrect indentation level 4, expected level should be 6. [Indentation]
Audit done.
```

we're agreeing with formatter

**Comment by romani** [comment]
@rnveach, please review this PR, 
We need this to finalize migration phase of project.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 2 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:23]: Top portion is missing finally, excess indentation and whitespace violations is making it harder to review, but it is there, so I am good.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java (hunk 4 above)
Comment (rnveach) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocksNoViolations.java:61]: That is what I was looking for which wasn't provided.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocks.java [hunk 5]
```diff
@@ -1,60 +1,91 @@
 package com.google.checkstyle.test.chapter4formatting.rule413emptyblocks;
 
-class InputEmptyBlocksAndCatchBlocks
-{
-    static {}
```
Comment (rnveach) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocks.java:5]: If formatter changed this, then it seems like it is wrong.
https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.1.3-braces-empty-blocks
Also our google config wants a space between `{` and `}` which violate the same rule and the examples shown.

````
$ cat TestClass.java
/** Some javadoc. */
public class TestClass {
  static {} // violation

  static { } // no violation
}

$ java -jar checkstyle-10.17.0-all.jar -c /google_checks.xml TestClass.java
Starting audit...
[WARN] TestClass.java:3:10: WhitespaceAround: '{' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] TestClass.java:3:11: WhitespaceAround: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.
````

**Review by rnveach — APPROVED** [review]

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocks.java (hunk 5 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyBlocksAndCatchBlocks.java:5]: Moved to https://github.com/checkstyle/checkstyle/issues/15424 to investigate separately
